internal/capabilities/detector.go

```diff
@@ -0,0 +1,100 @@
+package capabilities
+
+import (
+	"strings"
+
+	"github.com/masa-finance/tee-worker/api/types"
+)
+
+// JobServerInterface defines the methods we need from JobServer to avoid circular dependencies
+type JobServerInterface interface {
+	GetWorkerCapabilities() map[string][]string
+}
+
+// DetectCapabilities automatically detects available capabilities based on configuration
+// If jobServer is provided, it will use the actual worker capabilities
+func DetectCapabilities(jc types.JobConfiguration, jobServer JobServerInterface) []string {
+	var detected []string
+	
+	// If we have a JobServer, get capabilities directly from the workers
+	if jobServer != nil {
+		workerCaps := jobServer.GetWorkerCapabilities()
+		for _, caps := range workerCaps {
+			detected = append(detected, caps...)
+		}
+		return detected
+	}
+	
+	// Fallback to basic detection if no JobServer is available
+	// This maintains backward compatibility and is used during initialization
+	
+	// Always available capabilities
+	detected = append(detected, "web-scraper", "telemetry", "tiktok-transcription")
+	
+	// Check for Twitter capabilities based on credentials
+	if accounts, ok := jc["twitter_accounts"].([]string); ok && len(accounts) > 0 {
+		// Basic Twitter capabilities when accounts are available
+		detected = append(detected, "searchbyquery", "getbyid", "getprofilebyid")
```

Comment:
as a TODO item for the future: instead of using just strings, let's use stronger typing. As in:

```
type Capability string

const (
  SearchByQuery Capability = "searchbyquery"
  GetById Capability = "getbyid"
  // and so on
)
```

If we then add them to `tee-types` we can e.g. have a `capability.Match(jt JobType)` method which takes into account wildcards. This needs a bit more design, is it `capability.Match(jt JobType)` or `jobType.Match(capabilities ...Capability)`. I think the second is more ergonomic.

This makes APIs more self-documenting, as well as preventing misspells.

And let's keep this pattern in mind for the future, please.